src/swaps/bitcoin.rs

```diff
@@ -1205,16 +1205,24 @@ impl BtcSwapTx {
     }
 
     /// Calculate the size of a transaction.
+    /// The `preimage` is only required when calculating the claim tx size.
     /// Use this before calling drain to help calculate the absolute fees.
     /// Multiply the size by the fee_rate to get the absolute fees.
-    pub fn size(&self, keys: &Keypair, preimage: &Preimage) -> Result<usize, Error> {
-        let dummy_abs_fee = 0;
-        // Can only calculate non-coperative claims
+    pub fn size(
+        &self,
+        keys: &Keypair,
+        preimage: Option<&Preimage>,
```

Comment:
The preimage is always 32 bytes. Imho there is no need to actually pass the real value